diff --git a/CMakeLists.txt b/CMakeLists.txt
index c715bb20..0fd4a56d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,7 +24,8 @@ if(SOPHUS_MASTER_PROJECT)
     set(CMAKE_BUILD_TYPE Release)
   endif()
 
-  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD 20)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON) # CMake will generate an error if it is unable to compile the code with a compiler that supports C++17.  
 
   # Set compiler specific settings (FixMe: Should not cmake do this for us automatically?)
   IF("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
@@ -34,7 +35,7 @@ if(SOPHUS_MASTER_PROJECT)
   ELSEIF("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     SET(CMAKE_CXX_FLAGS_DEBUG  "-O0 -g")
     SET(CMAKE_CXX_FLAGS_RELEASE "-O3")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wextra -std=c++14 -Wno-deprecated-declarations -ftemplate-backtrace-limit=0")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wextra -Wno-deprecated-declarations -ftemplate-backtrace-limit=0")
     SET(CMAKE_CXX_FLAGS_COVERAGE "${CMAKE_CXX_FLAGS_DEBUG} --coverage -fno-inline -fno-inline-small-functions -fno-default-inline")
     SET(CMAKE_EXE_LINKER_FLAGS_COVERAGE "${CMAKE_EXE_LINKER_FLAGS_DEBUG} --coverage")
     SET(CMAKE_SHARED_LINKER_FLAGS_COVERAGE "${CMAKE_SHARED_LINKER_FLAGS_DEBUG} --coverage")
diff --git a/sophus/so3.hpp b/sophus/so3.hpp
index 67dd852f..94b41ab6 100644
--- a/sophus/so3.hpp
+++ b/sophus/so3.hpp
@@ -455,6 +455,53 @@ class SO3Base {
     return static_cast<Derived const*>(this)->unit_quaternion();
   }
 
+  // newly-added
+  // left jacobian
+  SOPHUS_FUNC static Transformation jl(const Point& Omega) {
+    using std::cos;
+    using std::sin;
+
+    Scalar theta = Omega.norm();
+    if (theta < Constants<Scalar>::epsilon()) {
+      return Transformation::Identity() + Scalar(0.5) * SO3<Scalar>::hat(Omega);
+    }
+
+    Point a = Omega;
+    a.normalize();
+    Scalar sin_theta = sin(theta);
+    Scalar cos_theta = cos(theta);
+    return (sin_theta / theta) * Transformation ::Identity() +
+           (1 - sin_theta / theta) * a * a.transpose() +
+           (1 - cos_theta) / theta * SO3<Scalar>::hat(a);
+  }
+
+  SOPHUS_FUNC static Transformation jl_inv(const Point& Omega) {
+    Scalar theta = Omega.norm();
+    if (theta < 1e-6) {
+      return Transformation::Identity();
+    }
+
+    Point a = Omega;
+    a.normalize();
+
+    double cot_half_theta = cos(0.5 * theta) / sin(0.5 * theta);
+    return 0.5 * theta * cot_half_theta * Transformation ::Identity() +
+           (1 - 0.5 * theta * cot_half_theta) * a * a.transpose() -
+           0.5 * theta * SO3<Scalar>::hat(a);
+  }
+
+  SOPHUS_FUNC static Transformation jr(const Point& Omega) {
+    return jl(-Omega);
+  }
+  SOPHUS_FUNC Transformation jr() { return jr(log()); }
+  SOPHUS_FUNC Transformation jr_inv() { return jl_inv(-log()); }
+  SOPHUS_FUNC static Transformation jr_inv(const Point& Omega) {
+    return jl_inv(-Omega);
+  }
+  SOPHUS_FUNC static Transformation jr_inv(const SO3Base<Derived>& R) {
+    return jr_inv(R.log());
+  }
+
  private:
   /// Mutator of unit_quaternion is private to ensure class invariant. That is
   /// the quaternion must stay close to unit length.
